pkg/plugin/driver/daemonset/daemonset.go

```diff
@@ -28,34 +28,41 @@ import (
 	v1 "k8s.io/api/core/v1"
 	v1beta1ext "k8s.io/api/extensions/v1beta1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-
 	kuberuntime "k8s.io/apimachinery/pkg/runtime"
+
 	"k8s.io/client-go/kubernetes"
 	"k8s.io/client-go/kubernetes/scheme"
 )
 
 // Plugin is a plugin driver that dispatches containers to each node,
 // expecting each pod to report to the master.
 type Plugin struct {
-	Definition      plugin.Definition
-	DfnTemplateData *plugin.DefinitionTemplateData
-	cleanedUp       bool
+	Definition plugin.Definition
+	SessionID  string
+	Namespace  string
+	cleanedUp  bool
 }
 
 // Ensure DaemonSetPlugin implements plugin.Interface
 var _ plugin.Interface = &Plugin{}
 
+type templateData struct {
+	PluginName        string
+	ResultType        string
```

Comment:
Do we even use result type anymore?  Originally it was file extension and now it's url mundge = plugin name.

Reply:
it's used separately from the name in the worker. I don't know if anyone defines them separately, but certainly they _could_